lucene/core/src/java/org/apache/lucene/index/IndexWriter.java

```diff
@@ -1009,6 +1009,14 @@ public IndexWriter(Directory d, IndexWriterConfig conf) throws IOException {
         changed();
 
       } else if (reader != null) {
+        if (reader.segmentInfos.getIndexCreatedVersionMajor() < Version.MIN_SUPPORTED_MAJOR) {
```

Comment:
Hmm does `addIndexes` try to verify version of the incoming index is not too old?  We will keep doing that, right?  I.e. the only added best effort here is when directly opening an `IndexReader` you can (with this change) now ask that older versions be allowed.

Reply:
yes we verify that it's the same major as the index we are adding to.